### PR TITLE
Refactor the logic for drawing selected mobile sessions.

### PR DESCRIPTION
**Pull request by AkkarinForest**
It's just a first step in making it more readable. I will appreciate any suggestions on how to improve it further. Also, it fixes two bugs:
- mobile: session measurements are not updating colors according to heat level changes when session selected
- mobile: blue line doesn't disappear after deselecting a session

### CODE REVIEW

File: app/assets/javascripts/code/services/google/_map.js [hunk 1]
```diff
@@ -135,36 +135,12 @@ export const map = (
...
-      var latlng = new google.maps.LatLng(latLngObj.latitude, latLngObj.longitude);
-      var icon = "/assets/location_marker0.png";
+    drawMarker: function({ position, title, zIndex, icon }){
+      var newMarker = new google.maps.Marker({ position, title, zIndex, icon });
 
-      if (level) {
-        icon = "/assets/location_marker" + level + ".png";
-      }
+      newMarker.setMap(this.get());
+      this.markers.push(newMarker);
```
Comment (AkkarinForest) [inline app/assets/javascripts/code/services/google/_map.js]: I would like to add test for that but I'm not sure how to mock `new google.maps.Marker`?

File: app/assets/javascripts/tests/helpers.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ export const mock = name => {
   return {
     [name]: (arg1, arg2, arg3) => { calls1.push(arg1); calls2.push(arg2); calls3.push(arg3); },
     wasCalled: () => calls1.length === 1,
+    wasCalled3Times: () => calls1.length ===3,
```
Comment (3v0k4) [inline app/assets/javascripts/tests/helpers.js]: Have you considered refactoring `wasCalled` to be used like `wasCalled({ times: 3 })` or introducing `wasCalledNTimes(3)`?

File: app/assets/javascripts/tests/_draw_session.test.js [hunk 3]
```diff
@@ -16,8 +16,47 @@ test('clearOtherSessions removes all markers expect the selected one', t => {
   t.end();
 });
 
+test('drawMobileSession draws a session when session is loaded and sensor is selected', t => {
+  const map = mock('drawMarker');
+  const heat = { getLevel: () => 1 };
```
Comment (3v0k4) [inline app/assets/javascripts/tests/_draw_session.test.js]: Does this influence the test output? Example if it returned 0 the output would be !=? Maybe you could make that more explicit or hide it if it's noisy

File: app/assets/javascripts/tests/_draw_session.test.js [hunk 4]
```diff
@@ -16,8 +16,47 @@ test('clearOtherSessions removes all markers expect the selected one', t => {
...
+  const sensors = selectedSensor;
```
Comment (3v0k4) [inline app/assets/javascripts/tests/_draw_session.test.js]: Have you considered inlining this in the obj?

Comment (3v0k4) [reply app/assets/javascripts/tests/_draw_session.test.js]: Have you considered not passing `sensors` in `_drawSession` and have a default inside `_drawSession`?

File: app/assets/javascripts/code/services/google/graph_highlight.js [hunk 5]
```diff
@@ -15,7 +15,11 @@ angular.module("google").factory("graphHighlight",  ["map",
       this.hide();
       var self = this;
       _(points).each(function(point){
-        self.items.push({marker: map.drawMarker(point), point: point});
+        self.items.push({marker: map.drawMarker({
```
Comment (3v0k4) [inline app/assets/javascripts/code/services/google/graph_highlight.js]: Have you considered extracting something here? It seems a bit noisy

File: app/assets/javascripts/code/services/google/_map.js (hunk 1 above)
Comment (3v0k4) [reply app/assets/javascripts/code/services/google/_map.js]: Ping me for a chat later on about this if you want ;)

File: app/assets/javascripts/code/services/_draw_session.js [hunk 6]
```diff
@@ -95,3 +96,17 @@ export const drawSession = (
 };
 
 const calculateHeatLevel = (heat, value) => heat.getLevel(value);
+
+const createMeasurementMarker = (measurement, idx, heat, map, suffix) => {
+  var roundedValue = Math.round(measurement.value);
+  var level = calculateHeatLevel(heat, roundedValue);
+  if (level){
```
Comment (3v0k4) [inline app/assets/javascripts/code/services/_draw_session.js]: I'd suggest an early return here. It makes more clear what happens when level is falsy and keeps code linear

File: app/assets/javascripts/code/services/google/graph_highlight.js (hunk 5 above)
Comment (AkkarinForest) [reply app/assets/javascripts/code/services/google/graph_highlight.js]: Yes! But I don't want to move too many stuff at once or the whole messed up ecosystem might collapse. I'm leaving graph_highlight.js and note.js for the next round :)

**Pull request merged**